### PR TITLE
Do not trash preserved registers (like 'ebp') when hell node processing

### PR DESCRIPTION
**Pull request by ptomin**
- Use `Platform.CreateTrashedRegisters()` to get trashed register when hell node processing
- Add stack register to list of trashed registers on `win32` and `msdos`
- Output for 7 unit tests was changed. These tests uses `msdos` platform. Some of 86 already broken tests could be affected too, of course.

### CODE REVIEW

File: src/Decompiler/Analysis/SsaTransform.cs [hunk 1]
```diff
@@ -385,6 +387,14 @@ private void GenerateUseDefsForUnknownCallee(CallInstruction ci)
             }
         }
 
+        private bool IsTrashed(Storage stg)
+        {
+            if (!(stg is RegisterStorage) || stg is TemporaryStorage)
+                return false;
+            return program.Platform.CreateTrashedRegisters()
+                .Where(r => r.OverlapsWith(stg)).Any();
+        }
```
Comment (uxmal) [inline src/Decompiler/Analysis/SsaTransform.cs]: Calling `Platform.CreateTrashedRegisters()` is only needed once; you could do that outside of this functoin and pass it in as a parameter.

File: src/Environments/Windows/Win32Platform.cs [hunk 2]
```diff
@@ -106,6 +106,7 @@ public override HashSet<RegisterStorage> CreateTrashedRegisters()
                 Registers.eax,
                 Registers.ecx,
                 Registers.edx,
+                Registers.esp,
             };
```
Comment (uxmal) [inline src/Environments/Windows/Win32Platform.cs]: Don't forget the `Win16Platform`, which is similar to `MsdosPlatform`

**Comment by uxmal** [comment]
Gah! Sorry for the merge conflict; didn't see your PR until after I pushed. I added some comments, please address them and then I'll pull the PR.

**Comment by ptomin** [comment]
Never mind! I'll do rebase

File: src/Decompiler/Analysis/SsaTransform.cs (hunk 1 above)
Comment (ptomin) [inline src/Decompiler/Analysis/SsaTransform.cs]: Done in 44f49bb

File: src/Environments/Windows/Win32Platform.cs (hunk 2 above)
Comment (ptomin) [inline src/Environments/Windows/Win32Platform.cs]: Done in 85a2f9f

**Comment by ptomin** [comment]
> I added some comments, please address them

I have passed trashed registers as a parameter of `IsTrashed` and added `sp` to trashed registers list of `win16`. Anything else?

**Comment by ptomin** [comment]
Good. Now I'm going to inject `IndirectCallRewriter` to analysis so that it works on `VCExeSample`

**Pull request merged**